src/OpenLoco/Windows/BuildVehicle.cpp

```diff
@@ -482,13 +483,13 @@ namespace OpenLoco::Ui::Windows::BuildVehicle
                 }
             }
 
-            auto power = std::min<uint16_t>(vehicleObj->power, 1);
-            // Unsure why power is only checked for first byte.
-            buildableVehicles.push_back({ vehicleObjIndex, static_cast<uint8_t>(power), vehicleObj->designed });
+            // 'power' is a flag indicating if the vehicle is powered or not. 1 = powered, 0 = unpowered
+            auto power = static_cast<uint8_t>(std::min<uint16_t>(vehicleObj->power, 1));
```

Comment:
should we change this to a bool? `isPowered`? Tbh i think it should just compare the whole of the power value may as well order it completely by power.

Reply:
`isPowered` is a good idea, I'll implement that tomorrow.

I'm just thinking about the 'final version' of the build vehicles window, and I can see one obvious set of enhancements is to add the table-sorting column headers, like OpenTTD does for its 'build vehicles' window:
![image](https://user-images.githubusercontent.com/7483209/159489309-2ea86388-a456-4039-873f-8257b805db69.png)

With those changes, there wouldn't even be a need to sort by date-then-power as you'd be able to filter out vehicles by type and then sort by an individual statistic.

Reply:
Hmm. I'm a little apprehensive about adding table/sort header buttons to lists that don't have obvious columns, to be honest. I think the OpenTTD screenshot isn't very intuitive UI.

Reply:
Sure, I'm not saying copy the OpenTTD UI, I'm saying copy the functionality.
Being able to only display vehicles that are powered, or vehicles that carry passengers, etc, would be an immensely useful feature in my opinion.